integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
+        try {
+
+            SparkContext sparkContext = SparkContext.getOrCreate();
+            hiveContext = new org.apache.spark.sql.hive.HiveContext(sparkContext);
+
+            String query = "SELECT COUNT(*) FROM " + databaseName
+                           + "."
+                           + tableName
+                           + " ";
+
+            if (whereClause != "") {
+                query = query + "WHERE " + whereClause;
+            }
+
+            log.info("Executing hive query: " + query);
+
+            DataFrame countDF = hiveContext.sql(query);
+
+            // Only take the first value which contains the row count
+            Long rowCount = countDF.collect()[0].getLong(0);
+
+            return rowCount;
+
+        } catch (Exception e) {
+            log.error("ERROR - Error while getting row count. Parameters were " +
+                      " database = "
+                      + databaseName
+                      +
+                      " table = "
+                      + tableName
+                      +
+                      " whereClause = "
+                      + whereClause, e);
+
+            throw e;
+        }
     }
 
-    public void doDataQualityChecks() {
+    /**
+     * Write the results to a Hive table
+     */
+    private void outputToHive() {
+
         try {
+            String databaseName = flowAttributes.getAttributeValue(DataQualityConstants.CATEGORY_ATTRIBUTE);
+            String tableName = flowAttributes.getAttributeValue(DataQualityConstants.FEED_ATTRIBUTE);
+            String dqTableName = tableName + DataQualityConstants.DQ_TABLE_SUFFIX;
+            String processing_dttm = flowAttributes.getAttributeValue(DataQualityConstants.PROCESSING_DTTM_ATTRIBUTE);
+
             SparkContext sparkContext = SparkContext.getOrCreate();
             hiveContext = new org.apache.spark.sql.hive.HiveContext(sparkContext);
 
-            String sql = "SELECT COUNT(*) FROM " + categoryName + "." + feedName;
-            log.info("Executing query {}", sql);
-            DataSet dataFrame = scs.sql(getHiveContext(), sql);
-            List<Row> resultRows = dataFrame.collectAsList();
-            long rowCount = 0;
-            if (resultRows.size() > 0) {
-                rowCount = resultRows.get(0).getLong(0);
+            DataQualityWriter dqWriter = DataQualityWriter.getInstance();
+            dqWriter.setHiveContext(hiveContext);
+            dqWriter.setSparkContext(new JavaSparkContext(sparkContext));
+
+            for (DataQualityRule rule : ruleList) {
+                dqWriter.addRow(new DataQualityRow(rule));
             }
-            log.info("Total rows in {}.{}: {}", categoryName, feedName, rowCount);
+
+            dqWriter.writeResultToTable(scs,
+                                        databaseName + "." + dqTableName,
+                                        processing_dttm);
+
+        } catch (MissingAttributeException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)